Unit test test_data_validation.py fails due to lazy loading of tabulate package

Unit test module `test_data_validation.py` fails when run in isolation, this does not impact production code:

```
pytest tests/unit/test_data_validation.py -k test_data_validation_client
...
tests/unit/test_data_validation.py F
...
/.../data_validation/result_handlers/text.py:51: in get_formatted
    return result_df.drop(cols_filter_list, axis=1).to_markdown(
/.../.venv/lib/python3.12/site-packages/pandas/util/_decorators.py:333: in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
/.../.venv/lib/python3.12/site-packages/pandas/core/frame.py:2994: in to_markdown
    tabulate = import_optional_dependency("tabulate")
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/.../.venv/lib/python3.12/site-packages/pandas/compat/_optional.py:151: in import_optional_dependency
    if version and Version(version) < Version(minimum_version):
                   ^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'Version' object has no attribute '_version'") raised in repr()] Version object at 0x7f6da0bc2420>, version = 'unknown'

    def __init__(self, version: str) -> None:
        # Validate the version and parse it into pieces
        match = self._regex.search(version)
        if not match:
>           raise InvalidVersion(f"Invalid version: '{version}'")
E           pandas.util.version.InvalidVersion: Invalid version: 'unknown'

/.../.venv/lib/python3.12/site-packages/pandas/util/version/__init__.py:343: InvalidVersion
```

But passes when executed as part of `pytest tests/unit`.

### How to Fix It

Import tabulate at the module level in `test_data_validation.py` (or in `text.py`) so that tabulate is loaded before `pyfakefs` mocks the filesystem:
```
  import tabulate
```
When imported before pyfakefs activates, tabulate reads its metadata from the real filesystem (__version__ = "0.10.0"), allowing `pandas.to_markdown()` to succeed during test execution.

--- a/setup.py
+++ b/setup.py
@@ -59,6 +59,8 @@ extras_require = {
         "pytest-cov",
         "pytest-rerunfailures",
         "pytest-timeout",
+        "snowflake-connector-python",
+        "snowflake-sqlalchemy",
     ],
 }
 

--- a/tests/unit/test_data_validation.py
+++ b/tests/unit/test_data_validation.py
@@ -12,16 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from datetime import datetime, timedelta
 import json
 import logging
+import random
+from unittest import mock
+
+from google.cloud import bigquery
+import ibis.expr.datatypes as dt
 import pandas
 import pytest
-import random
-from datetime import datetime, timedelta
-from unittest import mock
-from google.cloud import bigquery
-
-import ibis.expr.datatypes as dt
+# Import tabulate here because it is lazily loaded in pandas which fails when fakefs is in play.
+import tabulate  # noqa: F401
 
 from data_validation import consts
 from data_validation.result_handlers.bigquery import BQRH_NO_WRITE_MESSAGE

--- a/tests/unit/test_data_validation.py
+++ b/tests/unit/test_data_validation.py
@@ -22,6 +22,7 @@ from google.cloud import bigquery
 import ibis.expr.datatypes as dt
 import pandas
 import pytest
+
 # Import tabulate here because it is lazily loaded in pandas which fails when fakefs is in play.
 import tabulate  # noqa: F401
 

